### PR TITLE
Feature/alarmvalues table

### PR DESCRIPTION
**Pull request by ALuesink**
Edited function for alarmvalues table at the begin of violinplots. These changes make it possible to have different alarmvalues for each metabolite.

### CODE REVIEW

File: pipeline/scripts/15-dIEM_violin.R [hunk 1]
```diff
@@ -27,6 +27,7 @@ source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions
 source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/prepare_toplist.R")
 source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/create_violin_plots.R")
 source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/prepare_alarmvalues.R")
+source("/hpc/dbg_mz/development/DIMS_output_Helix_alarmvalues/pipeline/scripts/AddOnFunctions/get_patient_data_to_Helix.R")
```
Comment (rernst) [inline pipeline/scripts/15-dIEM_violin.R:30]: See comment in #82 about development paths

File: pipeline/scripts/AddOnFunctions/get_patient_data_to_Helix.R [hunk 2]
```diff
@@ -0,0 +1,31 @@
+get_patient_data_to_Helix <- function(metab_interest_sorted, metab_list_all){
```
[the remark this answers is not in the document]
Comment (ALuesink) [reply pipeline/scripts/AddOnFunctions/get_patient_data_to_Helix.R:1]: Yes, this is the same code as #82. The dataframe resulting from this function is needed twice in 15-dIEM_violin.R. Once to make the Helix output file, (#82 line 394) and once for the alarmvalues table (#83 line 397). When these two PR's (#82 & #83) will be merged only the code of #82 line 394 is needed.

File: pipeline/scripts/15-dIEM_violin.R (hunk 1 above)
Comment (ALuesink) [reply pipeline/scripts/15-dIEM_violin.R:30]: Agree, is changed.

File: pipeline/scripts/15-dIEM_violin.R [hunk 3]
```diff
@@ -392,7 +393,10 @@ if (violin == 1) { # make violin plots
       # for category Diagnostics, make list of metabolites that exceed alarm values for this patient
       # for category Other, make list of top highest and lowest Z-scores for this patient
       if (grepl("Diagnost", pdf_dir)) {
-        top_metab_pt <- prepare_alarmvalues(pt_name, metab_interest_sorted)
+        # get table that combines DIMS results with stofgroepen/Helix table
+        DIMS_Helix_table <- get_patient_data_to_Helix(metab_interest_sorted, metab_list_all)
```
[the remark this answers is not in the document]
Comment (ALuesink) [reply pipeline/scripts/15-dIEM_violin.R:397]: Agree, is changed.

File: pipeline/scripts/AddOnFunctions/get_patient_data_to_Helix.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (ALuesink) [reply pipeline/scripts/AddOnFunctions/get_patient_data_to_Helix.R]: Is changed

**Review by rernst — APPROVED** [review]

**Pull request merged**